lib/puma/server.rb

```diff
@@ -576,6 +576,7 @@ def lowlevel_error(e, env, status=500)
 
     def response_to_error(client, requests, err, status_code)
       status, headers, res_body = lowlevel_error(err, client.env, status_code)
+      res_body = ["Payload Too Large"] if status == 413
```

Comment:
Just double checking my understanding: So with this refactoring, the `lowlevel_error` is called (and potentially a user specified `lowlevel_error_handler`), which didn't happen before? In order to preserve the existing behaviour – the hard-coded response body `Payload Too Large` we set the body here

Question: Should we even call `lowlevel_error` at all when `status_code` is 413? If we didn't run the user specified `lowlevel_error_handler` before this refactoring, we probably shouldn't start doing it now.

A future (major?) Puma version could let the user specified `lowlevel_error_handler` to control the full response for the "Payload Too Large" scenario.

Reply:
What if we introduce `HttpParserError413` and raise that instead of `HttpParserError` (we already have `HttpParserError501`)? Then we can, in `client_error`, call `prepare_response` directly.

Reply:
Or is the `HttpParserError` raise at https://github.com/puma/puma/pull/3582/files#r1897406430 problematic, in that either "Payload Too Large" or some other error can happen?

Reply:
Is `@error_status_code` important or can we get by without it? I mean, can we pass the status code when we raise `HttpParserError`? I think that is a clearer interface.

Reply:
Re adding `HttpParserError413`, I'm not sure about the idea of having error classes for every possible request error.  That's why I added `@error_status_code`.  We might need to add something like `@error_status_message` or something like it.

I didn't want this PR to touch error handling, but once I got into it, it couldn't be avoided.

Reply:
How about changing the code to:
```ruby
def response_to_error(client, requests, err, status_code)
  # @todo remove sometime later
  if status_code == 413
    status = 413
    res_body = ["Payload Too Large"]
    headers = {}
  else
    status, headers, res_body = lowlevel_error(err, client.env, status_code)
  end
  prepare_response(status, headers, res_body, requests, client)
end
```

Reply:
Yeah that will do it

---

But we could also pass information via the error object? Instead of adding `@error_status_message`, `HttpParserError` could hold status code and the message.

Reply:
> But we could also pass information via the error object?

I didn't state one goal, of which this is the start.  I think 'ruby' error handling can be totally contained in the new method `Client#process_env_body`.  I'd like to contain it all within Client, except it will need the call to generate the response.